flake.nix

```diff
@@ -43,18 +44,12 @@
           '';
         };
         devShells.default =
-          with pkgs;
-          mkShell {
-            buildInputs = [
+          pkgs.mkShell {
+            packages = with pkgs; [
               bc
-              cargo
```

Comment:
This was importing Rust tools from nixpkgs which don't necessarily match what's in your toolchain. Now coming from the same toolchain that you're using to build the default package above